llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1107,13 +1107,30 @@ static void simplifyRecipe(VPRecipeBase &R, VPTypeAnalysis &TypeInfo) {
       return Def->replaceAllUsesWith(A);
 
     // Try to fold Not into compares by adjusting the predicate in-place.
-    if (isa<VPWidenRecipe>(A) && A->getNumUsers() == 1) {
+    auto CanFold = [&A](VPUser *U) {
+      return match(
+          U, m_CombineOr(m_Not(m_Specific(A)),
+                         m_Select(m_Specific(A), m_VPValue(), m_VPValue())));
+    };
+    if (isa<VPWidenRecipe>(A) && all_of(A->users(), CanFold)) {
       auto *WideCmp = cast<VPWidenRecipe>(A);
       if (WideCmp->getOpcode() == Instruction::ICmp ||
           WideCmp->getOpcode() == Instruction::FCmp) {
         WideCmp->setPredicate(
             CmpInst::getInversePredicate(WideCmp->getPredicate()));
```

Comment:
```suggestion
            WideCmp->getInversePredicate());
```

Would be helpful to sink this into the branches that perform the replacements?

Reply:
Perhaps worth adding the Invert flag to the m_Cmp matcher (see example in PatternMatch) that automatically flips the predicate.

Reply:
`getInversePredicate` doesn't exist on VPWidenRecipe. I'm not sure if it's worthwhile adding a method to it, it looks like this line is the only user of it.